### PR TITLE
Enable integration test for s2i to devfile conversion

### PR DESCRIPTION
**Pull request by adisky**
**What type of PR is this?**

 /kind feature
 
**What does does this PR do / why we need it**:
s2i to devfile conversion test was dependent on odo-init-image release, the image containing required changes has been released, I missed to enable the integration test.

**Which issue(s) this PR fixes**:

Fixes #?

**PR acceptance criteria**:

- [ ] Unit test 

- [x] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: pkg/devfile/parser/context/context.go [hunk 1]
```diff
@@ -75,6 +75,18 @@ func (d *DevfileCtx) Populate() (err error) {
 	return d.populateDevfile()
 }
 
+// SetAbsPath sets absolute file path for devfile
+func (d *DevfileCtx) SetAbsPath() (err error) {
+	// Get devfile absolute path
+	if d.absPath, err = util.GetAbsPath(d.relPath); err != nil {
+		return err
+	}
+	klog.V(4).Infof("absolute devfile path: '%s'", d.absPath)
```
Comment (mik-dass) [inline pkg/devfile/parser/context/context.go:84]: this should be `2` according to https://github.com/openshift/odo/pull/3876/files

Comment (adisky) [reply pkg/devfile/parser/context/context.go]: done.

File: pkg/odo/cli/utils/convert.go [hunk 2]
```diff
@@ -96,12 +96,12 @@ func NewConvertOptions() *ConvertOptions {
...
 func (co *ConvertOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
 
-	if !util.CheckPathExists(component.ConfigFilePath) {
+	if !util.CheckPathExists(filepath.Join(co.componentContext, component.ConfigFilePath)) {
 		return errors.New("this directory does not contain an odo s2i component, Please run the command from odo component directory to convert s2i component to devfile")
 	}
 
 	co.context = genericclioptions.NewContext(cmd)
-	co.componentContext = component.LocalDirectoryDefaultLocation
+	// co.componentContext = component.LocalDirectoryDefaultLocation
```
Comment (mik-dass) [inline pkg/odo/cli/utils/convert.go:104]: can we remove this?

File: tests/integration/component.go [hunk 3]
```diff
@@ -872,53 +872,53 @@ func componentTests(args ...string) {
...
+		JustBeforeEach(func() {
+			project = helper.CreateRandProject()
+			originalDir = helper.Getwd()
+			os.Setenv("ODO_EXPERIMENTAL", "true")
+		})
 
-			It("should convert s2i component to devfile component successfully", func() {
+		JustAfterEach(func() {
+			helper.DeleteProject(project)
+			helper.Chdir(originalDir)
```
Comment (mik-dass) [inline tests/integration/component.go:885]: I think we can remove this

File: pkg/odo/cli/utils/convert.go (hunk 2 above)
Comment (adisky) [reply pkg/odo/cli/utils/convert.go]: done.

File: tests/integration/component.go (hunk 3 above)
Comment (adisky) [reply tests/integration/component.go]: done

**Review by mik-dass — APPROVED** [review]
Works for me
/lgtm

**Pull request merged**